sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryElementJsonConverter.cs

```diff
@@ -399,5 +222,14 @@ private static bool TryGetNonLiteral<T>(JsonElement json, out DataFactoryElement
 
             return element != null;
         }
+
+        /// <summary>
+        /// Serialization is not supported for DataFactoryElementJsonConverter.
+        /// </summary>
+        /// <exception cref="NotSupportedException">Thrown when attempting to serialize an object.</exception>
+        public override void Write(Utf8JsonWriter writer, object? value, JsonSerializerOptions options)
```

Comment:
Isn't this a breaking change?

Reply:
you mean behavior-wise? We did the same thing for ResponseError
https://github.com/Azure/azure-sdk-for-net/blob/c3fb5ffbd8370c1ece98ee2fe1351348d53fe0aa/sdk/core/Azure.Core/src/ResponseError.Serialization.cs#L32-L36
I think the idea is, we would not like the user to use JsonSerializer.Serialize for them.

Reply:
Implemented MRW.Write https://github.com/Azure/azure-sdk-for-net/pull/50447/files#diff-f6084e425ea2efb2a7d0ec038ce399ba6b2457b8978aff4373424acc25c7f768R16 in this PR as well though.